### PR TITLE
Move helper methods to presenters

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

**Comment by jcoyne** [comment]
@cbeer are you suggesting I should restore all the existing helper methods and deprecate them?

**Comment by cbeer** [comment]
Just that we should deprecate them back in the 6.x branch.

**Comment by mejackreed** [comment]
I'm not sure I understand the implications or the reasons for this. Could that be popped into a comment somewhere? Or commit message?

**Comment by jcoyne** [comment]
@mejackreed Presenters are an object oriented approach to organizing view helpers.  Having objects allows us to encapsulate similar functionality which makes it easier to reason about how the methods interact with each other.  Read more http://nithinbekal.com/posts/rails-presenters/

**Comment by mejackreed** [comment]
Thanks @jcoyne . I just wanted the context of the change to live alongside it somewhere.

**Comment by cbeer** [comment]
What's the right pattern for an application to customize these presenters? When they were helpers, it was a simple matter of overriding in the e.g. `ApplicationHelper`.  Ought we move the hard-coded presenter names into configuration so applications can provide their own subclasses?

**Comment by jcoyne** [comment]
> Ought we move the hard-coded presenter names into configuration so applications can provide their own subclasses

Yes.

File: app/controllers/concerns/blacklight/catalog.rb [hunk 1]
```diff
@@ -122,6 +133,12 @@ def has_search_parameters?
     #
     # non-routable methods ->
     #
+    #
+    ##
+    # Override this method if you want to use a different presenter class
```
Comment (mejackreed) [inline app/controllers/concerns/blacklight/catalog.rb]: Or configure it in catalog controller config, no?

**Comment by jcoyne** [comment]
Benchmarks on master  (control):
```
$ ab -n 100 'http://localhost:3000/en?q=&search_field=default&per_page=50'
This is ApacheBench, Version 2.3 <$Revision: 1796539 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking localhost (be patient).....done


Server Software:
Server Hostname:        localhost
Server Port:            3000

Document Path:          /en?q=&search_field=default&per_page=50
Document Length:        138756 bytes

Concurrency Level:      1
Time taken for tests:   86.325 seconds
Complete requests:      100
Failed requests:        0
Total transferred:      13951500 bytes
HTML transferred:       13875600 bytes
Requests per second:    1.16 [#/sec] (mean)
Time per request:       863.249 [ms] (mean)
Time per request:       863.249 [ms] (mean, across all concurrent requests)
Transfer rate:          157.83 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:   786  863  76.6    837    1185
Waiting:      786  863  76.6    837    1185
Total:        786  863  76.6    837    1185

Percentage of the requests served within a certain time (ms)
  50%    837
  66%    854
  75%    875
  80%    901
  90%    992
  95%   1052
  98%   1083
  99%   1185
 100%   1185 (longest request)
```

On this branch:
```
$ ab -n 100 'http://localhost:3000/en?q=&search_field=default&per_page=50'
This is ApacheBench, Version 2.3 <$Revision: 1796539 $>
Copyright 1996 Adam Twiss, Zeus Technology Ltd, http://www.zeustech.net/
Licensed to The Apache Software Foundation, http://www.apache.org/

Benchmarking localhost (be patient).....done


Server Software:
Server Hostname:        localhost
Server Port:            3000

Document Path:          /en?q=&search_field=default&per_page=50
Document Length:        138417 bytes

Concurrency Level:      1
Time taken for tests:   82.643 seconds
Complete requests:      100
Failed requests:        0
Total transferred:      13917600 bytes
HTML transferred:       13841700 bytes
Requests per second:    1.21 [#/sec] (mean)
Time per request:       826.433 [ms] (mean)
Time per request:       826.433 [ms] (mean, across all concurrent requests)
Transfer rate:          164.46 [Kbytes/sec] received

Connection Times (ms)
              min  mean[+/-sd] median   max
Connect:        0    0   0.0      0       0
Processing:   753  826  57.7    810    1039
Waiting:      753  826  57.7    810    1039
Total:        754  826  57.7    810    1040

Percentage of the requests served within a certain time (ms)
  50%    810
  66%    822
  75%    833
  80%    839
  90%    906
  95%    990
  98%   1039
  99%   1040
 100%   1040 (longest request)
```

So, this branch looks a little faster (although I suspect this is within the margin of error).

**Pull request closed without merging**